src/messaging/ExchangeMgr.cpp

```diff
@@ -273,7 +277,7 @@ void ExchangeManager::OnMessageReceived(const PacketHeader & packetHeader, const
 
     // If we found a handler or we need to send an ack, create an exchange to
     // handle the message.
-    if (matchingUMH != nullptr || payloadHeader.NeedsAck())
+    if (matchingUMH != nullptr || payloadHeader.NeedsAck() || session.GetGroupId().HasValue())
```

Comment:
This change does not make sense here.  All incoming group-session messages are unsolicited.  If we have a message coming in, and its group-session, and we have no UMH for it, then we're done, no?  We do not have to create an exchange or whatnot.

In fact, for a group session we should skip this block even if `payloadHeader.NeedsAck()`, I would think.

Reply:
Removed the GroupId check